add srv and msg for naoqi_apps/launch/external_collision_avoidance.launch

kochigami opened the pull request:
These are srvs and msgs for external-collision avoidance APIs of ALMotion. (https://github.com/ros-naoqi/naoqi_bridge/pull/60)
1. same as #12 
   msg/Frames.msg and srv/GetChainClosestObstaclePosition.srv :  for `ALMotionProxy::getChainClosestObstaclePosition`
2. same as #11 
   msg/ChainClosestObstaclePosition.msg : for `Event: "ALMotion/Safety/ChainVelocityClipped"`
3. same as #10 
   srv/SetOrthogonalSecurityDistance.srv, srv/GetOrthogonalSecurityDistance.srv,   srv/SetTangentialSecurityDistance.srv and srv/GetTangentialSecurityDistance.srv
   I renamed OrthogonalSecurityDistance.srv and TangentialSecurityDistance.srv 
   These are for
   `ALMotionProxy::getOrthogonalSecurityDistance`, `ALMotionProxy::getTangentialSecurityDistance`, `ALMotionProxy::setOrthogonalSecurityDistance` and `ALMotionProxy::setTangentialSecurityDistance`
4. same as #6 
   msg/ExternalCollisionProtectionNames.msg, srv/GetExternalCollisionProtectionEnabled.srv and srv/SetExternalCollisionProtectionEnabled.srv: 
   for `ALMotionProxy::getExternalCollisionProtectionEnabled` and `ALMotionProxy::setExternalCollisionProtectionEnabled`

kochigami committed:
add srv and msg for naoqi_apps/launch/externalCollisionAvoidance.launch

k-okada commented:
Nice, one comment; I think removing existing message may influence many
users,
- when these deleted message has added
- which node using that

if it seems that message is already used in many users, I think we should
keep old message as well for a some time, if that message is just added
recently, I think we can remove that.

## 

◉ Kei Okada

2016-09-25 17:10 GMT+09:00 Kanae Kochigami notifications@github.com:

> These are srvs and msgs for external-collision avoidance APIs of ALMotion.
> 
>    1.
> 
>    same as #12 https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/12
>    msg/Frames.msg and srv/GetChainClosestObstaclePosition.srv : for
>    ALMotionProxy::getChainClosestObstaclePosition
>    2.
> 
>    same as #11 https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/11
>    msg/ChainClosestObstaclePosition.msg : for Event: "ALMotion/Safety/
>    ChainVelocityClipped"
>    3.
> 
>    same as #10 https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/10
>    srv/SetOrthogonalSecurityDistance.srv, srv/
>    GetOrthogonalSecurityDistance.srv, srv/SetTangentialSecurityDistance.srv
>    and srv/GetTangentialSecurityDistance.srv
>    I renamed OrthogonalSecurityDistance.srv and
>    TangentialSecurityDistance.srv
>    These are for
>    ALMotionProxy::getOrthogonalSecurityDistance, ALMotionProxy::
>    getTangentialSecurityDistance, ALMotionProxy::
>    setOrthogonalSecurityDistance and ALMotionProxy::
>    setTangentialSecurityDistance
>    4.
> 
>    same as #6 https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/6
>    msg/ExternalCollisionProtectionNames.msg, srv/
>    GetExternalCollisionProtectionEnabled.srv and srv/
>    SetExternalCollisionProtectionEnabled.srv:
>    for ALMotionProxy::getExternalCollisionProtectionEnabled and
>    ALMotionProxy::setExternalCollisionProtectionEnabled
> 
> ---
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19
> Commit Summary
> - add srv and msg for naoqi_apps/launch/externalCollisionAvoidance.
>   launch
> 
> File Changes
> - _A_ msg/ChainClosestObstaclePosition.msg
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-0
>   (3)
> - _A_ msg/ExternalCollisionProtectionNames.msg
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-1
>   (9)
> - _A_ msg/Frames.msg
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-2
>   (7)
> - _A_ srv/GetChainClosestObstaclePosition.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-3
>   (5)
> - _A_ srv/GetExternalCollisionProtectionEnabled.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-4
>   (4)
> - _A_ srv/GetOrthogonalSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-5
>   (5)
> - _A_ srv/GetTangentialSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-6
>   (5)
> - _D_ srv/OrthogonalSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-7
>   (5)
> - _A_ srv/SetExternalCollisionProtectionEnabled.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-8
>   (5)
> - _A_ srv/SetOrthogonalSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-9
>   (5)
> - _A_ srv/SetTangentialSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-10
>   (5)
> - _D_ srv/TangentialSecurityDistance.srv
>   https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19/files#diff-11
>   (5)
> 
> Patch Links:
> - https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19.patch
> - https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/19, or mute the
> thread
> https://github.com/notifications/unsubscribe-auth/AAeG3DvXglbFHXJ4MyoNN2copreeVkQfks5qtiyFgaJpZM4KF2UP
> .

kochigami commented:
@k-okada Thank you very much for your comment. Sorry for my lack of explanation. 

> I renamed OrthogonalSecurityDistance.srv and TangentialSecurityDistance.srv

when original `OrthogonalSecurityDistance.srv` and `TangentialSecurityDistance.srv` are added: about one year ago in https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/5 
which node uses these: `naoqi_externalCollisionAvoidance` in https://github.com/ros-naoqi/naoqi_bridge/pull/48 (closed before merged, this is the old version of this pull-request)

These are srv files for the node which was not merged. 
Therefore, I don't think renaming these affects many users. Maybe the current user is only me.

Karsten1987 commented on srv/GetTangentialSecurityDistance.srv:
```diff
@@ -0,0 +1,5 @@
+# Service for getting the tangential security distance
+
+# Empty response
```
sorry for the nitpick, but should that comment be `#Empty request` ?

kochigami replied:
Thank you very much for your review.
I modified GetOrthogonalSecurityDistance.srv, too.

Karsten1987 commented:
lgtm :+1:

kochigami commented:
Sorry, I found chain names definition in Chains section of http://doc.aldebaran.com/2-4/family/pepper_technical/bodyparts_pep.html .

Finally, I modified these things:
- Therefore, I created Chains.msg and used it in ChainClosestObstaclePosition.msg.
- renamed ChainClosestObstaclePosition.msg to ChainVelocityClipped.msg 
  for Event: "ALMotion/Safety/ChainVelocityClipped"
- move Arms.msg from another pull-request  #20 for GetChainClosestObstaclePosition.srv

suryaambrose commented:
Hi, 

This looks very good, except for one thing that I have some trouble with. 
The names you use for the Chains are the ones for Pepper. The Nao chains are different for the legs (http://doc.aldebaran.com/2-4/family/robots/bodyparts.html). 
As much as possible, the naoqi_bridge (and its messages) should not be robot-dependant. And if I wanted to use those messages and https://github.com/ros-naoqi/naoqi_bridge/pull/60 on a Nao, I think I might have some weird results when looking for clipped velocity, as it probably won't have been set (https://github.com/ros-naoqi/naoqi_bridge/pull/60/files#diff-083fa805fd176c3d74dac308c0b60c7fR208). 

If having something general for all robots is not possible, then let's keep it that way, as it is already really good. But if a little more thought could provide a good solution, that would be awesome.

suryaambrose reviewed (changes requested):
I think "Chains.msg" should not be dependant on the robot

kochigami committed:
rm Set/Get Orthogonal/Tangential SecurityDistance.srv because we can use Set/GetFloat.srv instead

kochigami committed:
Chains.msg: add NAO's chains

kochigami committed:
Frames.msg: remove comment

kochigami committed:
add MoveFailed.msg

kochigami commented:
modfied:
`Chains.msg`: add chains for NAO as an enum type

delete `SetOrthogonalSecurityDistance.srv`, `GetOrthogonalSecurityDistance.srv`, `SetTangentialSecurityDistance.srv` and `GetTangentialSecurityDistance.srv` because we can use `SetFloat.srv` and `GetFloat.srv` instead

add `MoveFailed.msg`

reference: 
(NAO's chains) http://doc.aldebaran.com/2-4/family/robots/bodyparts.html
http://doc.aldebaran.com/2-5/naoqi/motion/reflexes-external-collision-api.html#reflexes-external-collision-api

kochigami commented:
This is a new feature, which may cause unexpected trouble when it is merged.
I close this and improve this in my local.
I'm sorry to leave this for a long time.
